From 8d8431c3e8023adbdffea9782b6f2370bb367673 Mon Sep 17 00:00:00 2001
From: Adam Williamson <adam@blueradius.ca>
Date: Tue, 25 Jun 2024 11:19:53 -0700
Subject: [PATCH] TimeSeal.py: make IAC_WONT_ECHO a literal as telnetlib is
 gone (#2233) (#2235)

telnetlib was removed in Python 3.13. As we only used it to
create a bytestring from these constants, let's just make it a
literal instead.

Signed-off-by: Adam Williamson <awilliam@redhat.com>
---
 lib/pychess/ic/TimeSeal.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/pychess/ic/TimeSeal.py b/lib/pychess/ic/TimeSeal.py
index 0d09ff1e1..1e99b7218 100644
--- a/lib/pychess/ic/TimeSeal.py
+++ b/lib/pychess/ic/TimeSeal.py
@@ -1,6 +1,5 @@
 import asyncio
 import sys
-import telnetlib
 import random
 import time
 import platform
@@ -23,7 +22,9 @@
 ENCODELEN = len(ENCODE)
 G_RESPONSE = "\x029"
 FILLER = b"1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
-IAC_WONT_ECHO = b"".join([telnetlib.IAC, telnetlib.WONT, telnetlib.ECHO])
+# was: b"".join([telnetlib.IAC, telnetlib.WONT, telnetlib.ECHO])
+# but telnetlib was removed in Python 3.13
+IAC_WONT_ECHO = b"\xff\xfc\x01"
 
 _DEFAULT_LIMIT = 2**16
 

